pymeasure/instruments/heidenhain/nd287.py

```diff
@@ -0,0 +1,118 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2021 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+from pyvisa.errors import VisaIOError
+from pymeasure.instruments import Instrument
+
+
+class Nd287(Instrument):
+    """ Represents the Heidenhain ND287 position display unit used to readout and display absolute position measured by
+        Heidenhain encoders.
+    """
+    
+    position = Instrument.measurement(
+        "\x1BA0200", "Read the encoder's current position.",
+        get_process=lambda p: Nd287.position_proc(float(p.split("\x02")[-1])),
+    )
+
+    status = Instrument.measurement(
+        "\x1BA0800", "Read the encoder's status bar"
+    )
+
+    # additional configuration properties that cannot be controlled through software, but should be configurable
+    # to match the values on the local device. #
+    _units = "mm"
+    
+    @classmethod
+    def position_proc(cls, pos):
+        """ Apply the appropriate scaling factor to the position read-out from the encoder based on the value of the
+            units property.
+        """
+        if cls._units == "mm":
```

Comment:
Have you tested that this works correctly, i.e. you can switch units after creating an instance by using self.units?

Also, be aware of the caveat that, if you have more than one ND287 at the same time, I think they then can't have different settings. A proper fix of this will be (WIP, but slowly), see #250.

Reply:
I did test this and it worked (at least for one device). I believe that you are correct though that multiple ND287 devices could not have different settings. To address this, I've implemented the `position` property manually so that it can query the instance `_units` attribute to apply the appropriate scale factor. I will keep an eye on #250 though and perhaps this property can be updated when #250 is accepted.

Reply:
A solution for #250 is nearing completion now, so I'd wait until #376 is merged, and then use the new functionality here.

Reply:
In the meantime, we have activated a linter, so there are a couple of linting reports in the "Files changed" tab, which you could fix when you have an opportunity.

Reply:
Great to hear! A couple of linter issues have reappeared -- you can run flake8 locally (`flake8 .`) to check for issues (if your editor does not do that automatically)

Reply:
Do you still see the linter issues on your end? I believe my latest commit has addressed them. I ran flake8 locally and don't see any issues in the remote or local checks on my end anymore.

Reply:
Hm, I guess some UI state in the browser must have been stale? Anyway, nothing to see anymore, thanks!